packages/router/src/adapters/contract/contract.ts

```diff
@@ -28,9 +30,20 @@ export const getContractAddress = (chainId: number): string => {
   return nxtpContractAddress;
 };
 
+export const getOracleContractAddress = (chainId: number): string => {
+  const { config } = getContext();
+  const oracleContractAddress = config.chainConfig[chainId]?.priceOracleAddress;
+  if (!oracleContractAddress) {
+    throw new Error(`No price oracle contract exists for chain ${chainId}`);
```

Comment:
Can we make a named error for this? Use the structure of the other errors and put them in the domain.

Reply:
Yes, we can